kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -219,6 +267,105 @@
       });
     },
     methods: {
+      getSidebarInfo() {
+        if (this.lessonId) {
+          this.fetchLessonSiblings();
+        } else {
+          if (this.content.id) {
+            this.fetchSiblings();
+          }
+        }
+      },
+      /** Returns a promise which will get the progress for a set of nodes and map it back onto
+       *  the nodes, which are then set to this.viewResourcesContents.
+       *  @modifies this.viewResourcesContents - Sets it to the value of nodes, mapped with
+       *  progress, if found
+       *  @returns {Promise} - Just returns a meaningless Promise in case you want to chain
+       *  synchronously
+       */
+      progressPromise(nodes) {
+        // Avoid unnecessary fetches, set values, return empty Promise
+        if (!this.$store.getters.isUserLoggedIn) {
+          nodes.map(node => (node.progress = 0));
+          this.viewResourcesContents = nodes;
+          return Promise.resolve();
+        }
+        const getParams = {
+          content_ids: nodes.map(node => node.content_id),
+        };
+        return ContentNodeProgressResource.fetchCollection({ getParams }).then(progresses => {
+          this.viewResourcesContents = nodes
+            .map(node => {
+              const matchingProgress = progresses.find(p => p.content_id === node.content_id) || {
+                progress: 0,
+              };
+              node.progress = matchingProgress.progress;
+              return node;
+            })
+            .filter(node => node.content_id !== this.content.content_id);
+        });
+      },
+      /**
+       * Prepares a list of content nodes which are also in the same lesson as this.lessonId
+       * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
+       */
+      fetchLessonSiblings() {
```

Comment:
Can filter `fetchContentNodeProgress` by `lesson_id` to avoid watiing for the two promises to resolve here. https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/plugins/learn/assets/src/composables/useContentNodeProgress.js#L29

Thinking we should maybe be adding the content nodes to what the LessonResource returns to avoid having to do these chained fetches (but no need to do that in this PR!).